app/controllers/batch_contexts_controller.rb

```diff
@@ -16,7 +16,16 @@ def new
   end
 
   def create
-    @batch_context = BatchContext.new(batch_contexts_params)
+    params = batch_contexts_params
+
+    # allow the staging location to be a Globus URL
+    if params[:staging_location].start_with?('https://')
+      globus_dest = GlobusDestination.find_with_globus_url(params[:staging_location])
+      params[:staging_location] = globus_dest.staging_location if globus_dest
+      params[:globus_destination] = globus_dest
+    end
+
```

Comment:
There is some validation around `staging_location`: https://github.com/sul-dlss/pre-assembly/blob/main/app/models/batch_context.rb#L155-L176

Does it work as expected with a globus URL in that field?

Reply:
Yes, but only because that while submitted with the staging_location form field, it is swapped out for the actual location prior to the object getting created. See https://github.com/sul-dlss/pre-assembly/blob/t1188-form/app/controllers/batch_contexts_controller.rb#L21-L26